packages/node_modules/@node-red/runtime/lib/module/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright JS Foundation and other contributors, http://js.foundation
```

Comment:
I think we should move all this code (plus the template files) into a separate module - `@node-red/subflow-exporter` - and then add that as a dependency to the core. Had we discussed the design, I would have proposed this approach from the start.

Once we have got the encryption parts resolved, we can iterate in this area to pull it out into its own module. No point doing that before we have everything else sorted out.

Reply:
I am very sorry for the delay in response.
I have committed the update as per your comment except plugin related part.
However, signing the new EasyCLA will take some time because it needs to be reviewed by our organization.

> I was surprised to see the encryption work included here. When we discussed this feature on the July call, you did mention the possibility of including the encoding work, but my feedback was to do one step at a time as I still had big concerns over the practicality of the encoding design. If you had pressed me to review the PRs sooner I would have given that very same feedback again.

I apologize for it.   I thought I had told you about my work plan and progress, but it seems I did not convey it well.

> This PR is blocked until we resolve the encoding part of it, rather than being able to move forward with the bits we had agreed to on the July call.
> 
> The slides from the July call say the encoding work would be added via a plugin. Looking at the design notes and the code, I don't see anything about plugins mentioned.

The plugins mentioned in the slide were envisioned as additional features after the original design was implemented because I thought plugin mechanism needs some kind of update for supporting it.
However, I would like to consider incorporating specification by plugin in the initial implementation, as you suggest.

> The design requires a user to add lots of configuration to their settings file - with custom JavaScript code to do the encoding/decoding. I see you've changed the default settings file to include base64 encoding/decoding. We should not have functional code like that in the settings file. `base64` does not provide any protection and there is no benefit for a user to select that option. If we include it in the settings file, there will be users who pick that option (without properly understanding it) and other users who don't have that in the settings file (all existing users...) and they then are unable to import the subflow. That is going to cause a support issue when things don't work.
> 
> The reason for having it as a plugin is to avoid the user have to write (copy/paste) any code.

According to my understanding, the current plugin does not have a fixed loading order with other normal nodes.
So it may be loaded  after subflow module loading.  
I think it needs to be loaded first to provide decoding method of subflow.
Can you let me know your opinion on this?

> I have added other comments through the code.

Thank you for your comments.  I think I have addressed most of them in this update, but please point out any oddities.